### PR TITLE
Feature/ticpp2

### PR DESCRIPTION
**Pull request by petersteneteg**
Fixes #...

Changes proposed in this PR:
 * ...
 * ...
 * ...

This has been tested on: ...

> Thank you for contributing to the Inviwo repository! Please ensure the following: 
> - Code is error and warning free
> - Code follows the Inviwo coding guidelines: 
>   https://github.com/inviwo/inviwo/wiki/Coding-Conventions
> - New Code / Processors follows the Inviwo documentation guidelines: 
>   https://github.com/inviwo/inviwo/wiki/Documentation-guide
> - New Code / processors are tested in a unittest and / or regression test: 
>   https://github.com/inviwo/inviwo/wiki/Manual#testing
> 
> Please remember to specify the environment this code been compiled and tested on:
> Operating System / Compiler / IDE / QT / CMake / Python versions.

### CODE REVIEW

File: ext/ticpp/include/ticpp/base.h [hunk 1]
```diff
@@ -99,44 +139,49 @@ class TICPP_API TiXmlBase : public TiCppRC {
...
-        TIXML_ERROR_PARSING_DECLARATION,
-        TIXML_ERROR_DOCUMENT_EMPTY,
-        TIXML_ERROR_EMBEDDED_NULL,
-        TIXML_ERROR_PARSING_CDATA,
-        TIXML_ERROR_DOCUMENT_TOP_ONLY,
-
-        TIXML_ERROR_STRING_COUNT
+    static void EncodeString(const std::string_view str, std::string* out);
+
+    // Table that returs, for a given lead byte, the total number of bytes
```
Comment (martinfalk) [inline ext/ticpp/include/ticpp/base.h]:
```suggestion
    // Table that returns, for a given lead byte, the total number of bytes
```

File: ext/ticpp/include/ticpp/printer.h [hunk 2]
```diff
@@ -25,12 +30,19 @@
...
+        , simpleTextPrint{false}
+        , indent{streamPrint == TiXmlStreamPrint::No ? "    " : ""}
+        , lineBreak{streamPrint == TiXmlStreamPrint::No ? "\n" : ""} {
+        buffer.reserve(4096);
+    }
 
-    virtual bool VisitEnter(const TiXmlDocument& doc);
-    virtual bool VisitExit(const TiXmlDocument& doc);
+    virtual bool VisitEnter(const TiXmlDocument& doc) { return true; }
+    virtual bool VisitExit(const TiXmlDocument& doc) { return true; }
```
Comment (martinfalk) [inline ext/ticpp/include/ticpp/printer.h]:
```suggestion
    virtual bool VisitEnter(const TiXmlDocument& doc) override { return true; }
    virtual bool VisitExit(const TiXmlDocument& doc) override { return true; }
```

File: ext/ticpp/src/attribute.cpp [hunk 3]
```diff
@@ -5,184 +5,151 @@
...
+                               const allocator_type& alloc)
+    : name{_name, alloc}, value{_value, alloc}, prev{nullptr}, next{nullptr}, location{} {}
+
+TiXmlAttribute::~TiXmlAttribute() = default;
+
 const TiXmlAttribute* TiXmlAttribute::Next() const {
-    // We are using knowledge of the sentinel. The sentinel
-    // have a value or name.
-    if (next->value.empty() && next->name.empty()) return 0;
+    // We are using knowledge of the sentinel. The sentinel have a value or name.
```
Comment (martinfalk) [inline ext/ticpp/src/attribute.cpp]:
```suggestion
    // We are using knowledge of the sentinel. The sentinel must have a value or name.
```

File: ext/ticpp/src/attribute.cpp [hunk 4]
```diff
@@ -5,184 +5,151 @@
...
+    if (next->value.empty() && next->name.empty()) return nullptr;
     return next;
 }
 
 const TiXmlAttribute* TiXmlAttribute::Previous() const {
-    // We are using knowledge of the sentinel. The sentinel
-    // have a value or name.
-    if (prev->value.empty() && prev->name.empty()) return 0;
+    // We are using knowledge of the sentinel. The sentinel have a value or name.
```
Comment (martinfalk) [inline ext/ticpp/src/attribute.cpp]:
```suggestion
    // We are using knowledge of the sentinel. The sentinel must have a value or name.
```

File: include/inviwo/core/io/serialization/deserializer.h [hunk 5]
```diff
@@ -51,32 +51,61 @@
...
+};
+
+/**
+ * \class ContainerWrapper
+ */
 template <typename T, typename ValueGetter, typename IdGetter>
-class ContainerWrapper;
+    requires std::is_invocable_v<IdGetter, TiXmlElement&> &&
+             std::is_invocable_r_v<ContainerWrapperItem<T>, ValueGetter,
+                                   std::invoke_result_t<IdGetter, TiXmlElement&>, size_t>
```
Comment (martinfalk) [inline include/inviwo/core/io/serialization/deserializer.h:78]: Use named concept instead? The requires clause is repeated several times in this file.

File: include/inviwo/core/io/serialization/serializebase.h [hunk 6]
```diff
@@ -238,21 +153,11 @@ class IVW_CORE_API NodeSwitch {
      * \brief NodeSwitch helps track parent node during recursive/nested function calls.
      *
      * @param serializer reference to serializer or deserializer
-     * @param node //Parent (Ticpp Node) element.
+     * @param node // the node to switch to
```
Comment (martinfalk) [inline include/inviwo/core/io/serialization/serializebase.h:156]:
```suggestion
     * @param node the node to switch to
```

**Review by martinfalk — DISMISSED** [review]

File: include/inviwo/core/io/serialization/deserializer.h (hunk 5 above)
Comment (petersteneteg) [reply include/inviwo/core/io/serialization/deserializer.h:78]: ContainerWrapperItem  will be removed in a future commit

File: include/inviwo/core/io/serialization/serializebase.h (hunk 6 above)
Comment (petersteneteg) [reply include/inviwo/core/io/serialization/serializebase.h:156]: Will fix in a later PR

**Review by martinfalk — APPROVED** [review]

**Pull request merged**